core/src/main/java/org/apache/carbondata/core/indexstore/UnsafeMemoryDMStore.java

```diff
@@ -52,11 +62,23 @@
 
   private final long taskId = ThreadLocalTaskInfo.getCarbonTaskInfo().getTaskId();
 
-  public UnsafeMemoryDMStore(CarbonRowSchema[] schema) throws MemoryException {
+  private boolean addRowToUnsafe = true;
```

Comment:
Class name itself `UnsafeMemoryDMStore` so it does not make sense to have this variable.
Please create AbstractClass and give two implementations with safe and Unsafe.